src/api/schema/events/mod.rs

```diff
@@ -0,0 +1,32 @@
+mod log;
+
+use crate::event::EventInspector;
+use async_graphql::{Context, Subscription};
+use async_stream::stream;
+use log::LogEvent;
+use std::sync::Arc;
+use tokio::stream::Stream;
+
+#[derive(Default)]
+pub struct EventsSubscription;
+
+#[Subscription]
+impl EventsSubscription {
+    /// A stream of component log events
+    pub async fn log_events<'a>(
+        &'a self,
+        ctx: &'a Context<'a>,
+        component_name: String,
+    ) -> impl Stream<Item = LogEvent> + 'a {
+        stream! {
+            if let Some(mut receiver) = ctx
+                .data_unchecked::<Arc<EventInspector>>()
```

Comment:
This will panic if the `Arc<EventInspector>` isn't instantiated, but a check if already performed upstream before it gets to this point. It'd expect this to be marginally faster than a safe check that returns an `Option<Arc<EventInspector>>`.

Reply:
Note: This has now been renamed to `TapController`, but the same note applies.